pkg/shared/clients/redis/redis_client.go

```diff
@@ -49,6 +52,10 @@ func NewInClusterRedisClient() *RedisClient {
 			opts.Addrs = strings.Split(urls, ",")
 		}
 	}
+	if i, e := strconv.Atoi(os.Getenv(v1alpha1.EnvISBSvcRedisClusterMaxRedirects)); e == nil {
```

Comment:
No `else`: the default value is 3. 
https://github.com/numaproj/numaflow/pull/195/files/ee62ae5d7c15a26ccc55d91544cccea43919107f#diff-0cc47b7ab37a93ea2d8d21417a490806a15b89abc0ffdf192d4685be777b7283R41

We use the universal client in our shared redis client code. The `MaxRedirects` is only used for  [cluster mode](https://github.com/go-redis/redis/blob/3d1e2e5bc8af1bfec675334a9a3f4cd9919dfda3/universal.go#L51-L56)
ref:
- [ NewClusterClient()](https://github.com/go-redis/redis/blob/3d1e2e5bc8af1bfec675334a9a3f4cd9919dfda3/universal.go#L213)
- [cluster mode opts](https://github.com/go-redis/redis/blob/3d1e2e5bc8af1bfec675334a9a3f4cd9919dfda3/universal.go#L78)

If we don't set the [default `MaxRedirects` value](https://github.com/numaproj/numaflow/pull/195/files/ee62ae5d7c15a26ccc55d91544cccea43919107f#diff-0cc47b7ab37a93ea2d8d21417a490806a15b89abc0ffdf192d4685be777b7283R41) in universal opts, the `MaxRedirects` value is 0.
If `MaxRedirects` is set to 0, then it means we don't allow any redirects, and we will get `MOVED xxx xxx` error when redis redirects to another node to find the data.

> how this parameter will be configured (get it into RedisConfig?), and convert it into an ENV variable, etc. Can we hardcode it for now?

users don't need to provide the env since the default value is 3 and the var is only used for redis cluster mode.

Reply:
Set the default value to 3 because if we directly use the cluster client instead of universal client, the default value is 3
https://github.com/go-redis/redis/blob/master/cluster.go#L33-L36

Reply:
please document this